### PR TITLE
Added new metrics for Dicomcastworker

### PR DESCRIPTION
**Pull request by poadhika**
## Description
The goal here is to emit metrics when dicomcast  couldnot authenticate to dicomserver or fhirserver and add an alerting based on these metrics.

## Related issues
Addresses [issue #92890](https://microsofthealth.visualstudio.com/Health/_workitems/edit/92890)

## Testing
Tested manually with hosted service.

### CODE REVIEW

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs [hunk 1]
```diff
@@ -100,6 +109,23 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
         {
             LogUnhandledExceptionDelegate(_logger, ex);
 
+            if ((ex is FhirException) && ((FhirException)ex).StatusCode == HttpStatusCode.Forbidden)
+            {
+                _telemetryClient.GetMetric("Cast-To-Fhir-Forbidden").TrackValue(1);
```
Comment (jnlycklama) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs:114]: Would it make sense to keep these strings - `Cast-To-Fhir-Forbidden` , `Dicom-To-Cast-Forbidden` in a constants file or something? So they can be referenced from elsewhere, like in the unit tests

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core.UnitTests/Features/Worker/DicomCastWorkerTests.cs [hunk 2]
```diff
@@ -39,12 +43,18 @@ public DicomCastWorkerTests()
 
         _fhirService = Substitute.For<IFhirService>();
```
Comment (alishahahmed) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core.UnitTests/Features/Worker/DicomCastWorkerTests.cs:44]: Question for my understanding: why are _changeFeedProcessor and _hostApplication initialized outside of the constructor whereas _fhirService and _telemetryClient are initialized inside?

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/Constants.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.DicomCast.Core.Features.Worker;
+public static class Constants
+{
+    // Represents different metrics the dicomcastworker will emit.
+    public const string CastToFhirForbidden = "Cast-To-Fhir-Forbidden";
+    public const string DicomToCastforbidden = "Dicom-To-Cast-Forbidden";
```
Comment (alishahahmed) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/Constants.cs:11]: nit: `DicomToCastforbidden` -> `DicomToCastForbidden`

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs [hunk 4]
```diff
@@ -49,25 +54,29 @@ public class DicomCastWorker : IDicomCastWorker
     private readonly ILogger _logger;
     private readonly IHostApplicationLifetime _hostApplicationLifetime;
     private readonly IFhirService _fhirService;
+    private readonly TelemetryClient _telemetryClient;
```
Comment (alishahahmed) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs:57]: Where is this telemetry client registered?

Comment (alishahahmed) [reply converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs:57]: Probably it's in the PaaS side, but I just want to confirm.

Comment (poadhika) [reply converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs:57]: We  use AddApplicationInsightsTelemetry in both paas and startup class here to register this clinet.

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs [hunk 5]
```diff
@@ -100,6 +109,23 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
...
+                _telemetryClient.GetMetric(Constants.CastMIUnavailable).TrackValue(1);
+            }
+            else
+            {
+                _telemetryClient.GetMetric(Constants.CastingFailedForOtherReasons).TrackValue(1);
+            }
+
             // Any exception in ExecuteAsync will not shutdown application, call hostApplicationLifetime.StopApplication() to force shutdown.
             // Please refer to .net core issue on github for more details: "Exceptions in BackgroundService ExecuteAsync are (sometimes) hidden" https://github.com/dotnet/extensions/issues/2363
             _hostApplicationLifetime.StopApplication();
```
Comment (jnlycklama) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs:131]: Do we need to flush the telemetry client before the application is stopped? I needed to do that for ifx metrics, but not sure if Telemetry Client requires the same thing https://microsofthealth.visualstudio.com/Health/_git/workspace-platform?path=/dicom/app/Microsoft.Health.Cloud.Dicom.Web/Program.cs&version=GBpersonal/t-loemily/sendToEventGrid&line=36&lineEnd=37&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core.UnitTests/Features/Worker/DicomCastWorkerTests.cs (hunk 2 above)
Comment (poadhika) [reply converter/dicom-cast/src/Microsoft.Health.DicomCast.Core.UnitTests/Features/Worker/DicomCastWorkerTests.cs:48]: It was like this before. I updated the code such that they are initialized at the same place

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs (hunk 5 above)
Comment (poadhika) [reply converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs:131]: General practice is it is not required for the continuously running server application. 
https://docs.microsoft.com/en-us/java/api/com.microsoft.applicationinsights.telemetryclient.flush?view=azure-java-stable

**Review by jnlycklama — APPROVED** [review]

**Pull request merged**